src/config.c

```diff
@@ -0,0 +1,197 @@
+}
+
+static int setNumericValue(const char *name, RedisModuleString *value, void *privdata,
+                           RedisModuleString **err) {
+    RM_ConfigNumeric *config = privdata;
+    long long new_val;
+    if (RedisModule_StringToLongLong(value, &new_val) != REDISMODULE_OK) {
+        *err = RedisModule_CreateStringPrintf(NULL, "Invalid value for `%s`", name);
+        return REDISMODULE_ERR;
+    }
+    if (new_val < config->min || new_val > config->max) {
+        *err =
+            RedisModule_CreateStringPrintf(NULL, "Value for `%s` must be in the range [%lld, %lld]",
+                                           name, config->min, config->max);
+        return REDISMODULE_ERR;
+    }
+    config->value = new_val;
+
+    if (strcasecmp(name, "cf-bucket-size") == 0) {
+        rm_config.cf_initial_size.min = new_val * 2;
+    } else if (strcasecmp(name, "cf-initial-size") == 0) {
+        rm_config.cf_bucket_size.max = new_val / 2 < 255 ? new_val / 2 : 255;
+    }
+
+    return REDISMODULE_OK;
+}
+static RedisModuleString *getNumericValue(const char *name, void *privdata) {
+    return RedisModule_CreateStringPrintf(NULL, "%zu", *((size_t *)privdata));
+}
+
+typedef struct {
+    const char *name;
+    const char *defVal;
+    RedisModuleConfigSetStringFunc setValue;
+    RedisModuleConfigGetStringFunc getValue;
+    void *privdata;
+} RM_ConfigVar;
+
+RM_ConfigVar rm_config_vars[] = {
+    {
+        .name = "bf-error-rate",
+        .defVal = "0.01",
+        .setValue = setFloatValue,
+        .getValue = getFloatValue,
+        .privdata = &rm_config.bf_error_rate,
+    },
+    {
+        .name = "bf-initial-size",
+        .defVal = "100",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.bf_initial_size,
+    },
+    {
+        .name = "bf-expansion-factor",
+        .defVal = "2",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.bf_expansion_factor,
+    },
+    {
+        .name = "cf-bucket-size",
+        .defVal = "2",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_bucket_size,
+    },
+    {
+        .name = "cf-initial-size",
+        .defVal = "1024",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_initial_size,
+    },
+    {
+        .name = "cf-max-iterations",
+        .defVal = "20",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_max_iterations,
+    },
+    {
+        .name = "cf-expansion-factor",
+        .defVal = "1",
+        .setValue = setNumericValue,
+        .getValue = getNumericValue,
+        .privdata = &rm_config.cf_expansion_factor,
+    },
+    {
+        .name = "cf-max-expansions",
+        .defVal = "32",
```

Comment:
So if we have those defined as strings using the new config API, will the user receive those also as `"32"` in the output of `GET`? This **might** be confusing. Even though I understand the current problem with the API, I just was wondering if there is a way to specifically address this in particular?

Reply:
Even if we did solve this in the specific case, I don't really see a way around it in the general case. It's not any different than `CONFIG GET bf-error-rate` returning `"0.01"` and that has no workaround.